src/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/DiagnosticSourceEventSource.cs

```diff
@@ -860,48 +873,136 @@ public PropertySpec(string propertyName, PropertySpec? next = null)
                 /// </summary>
                 private class PropertyFetch
                 {
-                    protected PropertyFetch(Type type)
+                    public PropertyFetch(Type? type)
                     {
-                        Debug.Assert(type != null);
                         Type = type;
                     }
 
-                    internal Type Type { get; }
+                    /// <summary>
+                    /// The type of the object that the property is fetched from. For well-known static methods that
+                    /// aren't actually property getters this will return null.
+                    /// </summary>
+                    internal Type? Type { get; }
 
                     /// <summary>
-                    /// Create a property fetcher from a .NET Reflection PropertyInfo class that
-                    /// represents a property of a particular type.
+                    /// Create a property fetcher for a propertyName
                     /// </summary>
-                    public static PropertyFetch? FetcherForProperty(Type type, PropertyInfo? propertyInfo)
+                    public static PropertyFetch? FetcherForProperty(Type? type, string propertyName)
                     {
-                        if (propertyInfo == null)
+                        if (propertyName == null)
                             return new PropertyFetch(type);     // returns null on any fetch.
+                        if (propertyName == "*Activity")
+                            return new CurrentActivityPropertyFetch();
 
-                        var typedPropertyFetcher = typeof(TypedFetchProperty<,>);
-                        var instantiatedTypedPropertyFetcher = typedPropertyFetcher.GetTypeInfo().MakeGenericType(
-                            propertyInfo.DeclaringType!, propertyInfo.PropertyType);
-                        return (PropertyFetch?)Activator.CreateInstance(instantiatedTypedPropertyFetcher, type, propertyInfo);
+                        if (type == null)
+                        {
+                            // if we get here it is a bug, type should only be null for the well-known
+                            // static fetchers that were already checked above
+                            Debug.Assert(type != null);
```

Comment:
`Debug.Fail("Type should only be null for the well-known static fetchers already checked");`

Reply:
Or, since there's really no way this should happen, just delete the `if` block entirely and replace it with the above assert.